client/components/payment-country-select/index.jsx

```diff
@@ -33,19 +33,18 @@ export class PaymentCountrySelect extends Component {
 	componentDidMount = () => {
 		// Notify the callback function about the country (or lack thereof)
 		// that is pre-selected at the time the component is first displayed.
-		let countryCode = this.props.countryCode;
-		if ( ! filter( this.props.countriesList, { code: countryCode } ).length ) {
-			// If the stored payment country isn't one of the allowed countries
-			// for this component, pass along the default value instead (so
-			// that the value the callback function receives never represents a
-			// country that it considers to be an invalid choice, and so that
-			// it always matches what the user actually sees pre-selected in
-			// the form field).
-			countryCode = '';
-		}
-		this.props.onCountrySelected( this.props.name, countryCode );
+		this.props.onCountrySelected( this.props.name, this.props.countryCode );
 	};
 
+	componentDidUpdate( prevProps ) {
+		// In addition to 'componentDidMount' in case a valid countryCode
+		// is received on a later render not catched by 'handleFieldChange'
+		// i.e. a store update
+		if ( prevProps.countryCode !== this.props.countryCode ) {
+			this.props.onCountrySelected( this.props.name, this.props.countryCode );
+		}
```

Comment:
fun fact! React paid attention to common uses and created `getDerivedStateFromProps()` for situations like this were we are wanting to compare previous props to next props. in one function it handles the mount and update scenarios.

Reply:
I feel like `getDerivedStateFromProps` is not the right one here because it doesn't allow to compare with previous props by default. Other than `componentDidUpdate` it's receiving current props and if I understand it correctly we'd need to use state for a proper comparison.

Reply:
that's a correct inference. it changes how we compare to previous props in that we have to internalize them into state. at the same time it solves some other problems, notably here the duplication of the updating logic.

it's rather new and I won't say I fully understand it. just wanted to point it out 😄